serverless for iran

patterniha opened the pull request:
@RPRX @Fangliding

patterniha committed:
Add files via upload

patterniha commented:
All selected parameters have a reason behind them,
I can explain each one.

developer861 commented:
برای udp noise بهتر نیست delay و packet رو بازه بزارین؟
مثلا 
"delay":"1-5","packets":"1000-1400"

developer861 commented:
حجم 1232 برای نویز یکم زیاد نیست؟
به نظرم با کمترم راه میافته مثلا 100-200

Fangliding commented:
jsonc

patterniha commented:
> jsonc

why jsonc?

its a serverless config we don't have server config

patterniha commented:
> ?

aha, i thought c meant client.
ok

patterniha committed:
Update and rename serverless_for_iran.json to serverless_for_iran.jsonc

patterniha committed:
Update and rename serverless_with_mitm_for_iran.json to serverless_with_mitm_for_iran.jsonc

patterniha commented:
> ?

done.

patterniha commented:
> حجم 1232 برای نویز یکم زیاد نیست؟ به نظرم با کمترم راه میافته مثلا 100-200

ببین حداقل mtu ما برای ipv6 عدد 1280  هست معمولا که ۴۰ تاش برای ip-header میره ۸ تاشم برای udp-header .

و برای ipv4  هم این عدد اکیه رو روترهای کنونی.
پس نرم افزارا معمولا بسته هاشونو حداکثر ۱۲۳۲ بایتی میفرستن پس مام ۱۲۳۲ انتخاب میکنیم که هم فینگرپرینت نداشته باشه هم ماکسیمم شانس عبور از gfw  رو بده بعدشم فقط یه باره خیلی رو میزان ترافیک تاثیر نمیزاره

patterniha commented:
> برای udp noise بهتر نیست delay و packet رو بازه بزارین؟ مثلا "delay":"1-5","packets":"1000-1400"

ما حداکثر ۳۰ تا نویز میتونیم استفاده کنیم چون سرورها نویز بیشتر رو بلاک میکنند و به نظر من نیاز نیست چون اینطوری حتی میتونیم هر کدوم رو هم خواستیم تغییر بدیم

patterniha committed:
Update serverless_with_mitm_for_iran.jsonc

patterniha committed:
Update serverless_for_iran.jsonc

fix a bug

patterniha committed:
Update serverless_with_mitm_for_iran.jsonc

fix a bug

patterniha commented:
@Fangliding @RPRX 

i use "forceIP" in outbound with dialerProxy by mistake.
It had caused Happy-eye-ball to become inactive.
i fix this problem

also the ipv6 only problem doesn't happen anymore.

Please test both again.

patterniha committed:
Update serverless_for_iran.jsonc

change dns to opendns to work on all isp

patterniha committed:
Update serverless_with_mitm_for_iran.jsonc

change dns to open dns to works on all isp

patterniha committed:
Update serverless_for_iran.jsonc

patterniha committed:
Update serverless_for_iran.jsonc

change h2c to https+fragment for compatibility

patterniha committed:
Update README.md

patterniha committed:
Update serverless_for_iran.jsonc

patterniha committed:
Update serverless_with_mitm_for_iran.jsonc

patterniha committed:
Update serverless_for_iran.jsonc

add comments

patterniha committed:
Update serverless_with_mitm_for_iran.jsonc

add comments

developer861 commented:
tested works fine

patterniha committed:
Update serverless_for_iran.jsonc

patterniha committed:
Update serverless_with_mitm_for_iran.jsonc

patterniha committed:
Update serverless_for_iran.jsonc

some websites like https://account.live.com/ doesn't support
tls-fragment.
also currently tcp-fragment works on all iran isp.
so use tcp-fragment instead of tls-fragment.

patterniha committed:
Update serverless_with_mitm_for_iran.jsonc

some websites like https://account.live.com/ doesn't support 
tls-fragment
also currently tcp-fragment works on all iran isp
so use tcp-fragment instead of tls-fragment

patterniha committed:
Update serverless_with_mitm_for_iran.jsonc

patterniha commented:
some websites like https://account.live.com/ doesn't support tls-fragment
also currently tcp-fragment works on all iran isp
so i use tcp-fragment instead of tls-fragment in non-mitm config

patterniha committed:
Update serverless_for_iran.jsonc

patterniha committed:
Update serverless_with_mitm_for_iran.jsonc

patterniha committed:
Update serverless_for_iran.jsonc

patterniha committed:
Update serverless_with_mitm_for_iran.jsonc

patterniha committed:
Update serverless_for_iran.jsonc

block ads caused many websites to not load.
so removed

patterniha committed:
Update serverless_with_mitm_for_iran.jsonc

block ads caused many websites to not load 
so removed

patterniha committed:
Update README.md

patterniha committed:
Update serverless_for_iran.jsonc

patterniha committed:
Update serverless_with_mitm_for_iran.jsonc

patterniha committed:
Update serverless_for_iran.jsonc

remove iran.dat

patterniha committed:
Update serverless_with_mitm_for_iran.jsonc

remove iran.dat

patterniha committed:
Update README.md

patterniha commented:
Some users said that it is difficult for them to download the iran.dat file,
also, it is not possible to use it on some apps.

so i removed iran.dat,

The configs are very simple now.

patterniha commented:
I need to tune the fragment parameters ...

patterniha committed:
Update serverless_for_iran.jsonc

patterniha committed:
Update serverless_with_mitm_for_iran.jsonc

patterniha commented:
> I need to tune the fragment parameters ...

done.

patterniha committed:
Update serverless_for_iran.jsonc

reduce noise packet length, and set to random

patterniha committed:
Update serverless_with_mitm_for_iran.jsonc

reduce noise packet length, and set to random

patterniha committed:
Update serverless_for_iran.jsonc

patterniha committed:
Update serverless_with_mitm_for_iran.jsonc

RPRX commented:
你可以继续修改，明天可能要发新版，我到时合并

patterniha commented:
@RPRX 
thank you so much.
Are you still on your word To make me one of the channel's admin?
A lot of users send irrelevant messages and it needs to be managed

patterniha committed:
Update serverless_for_iran.jsonc

block ads again

patterniha committed:
Update serverless_with_mitm_for_iran.jsonc

block ads again

patterniha committed:
Update serverless_for_iran.jsonc

add tcp no delay

patterniha committed:
Update serverless_with_mitm_for_iran.jsonc

patterniha commented:
the hosts should be updated. 
instad of fallback address i should set WhatsApp address for Instagram and Facebook...

patterniha committed:
Update serverless_for_iran.jsonc

I finally decided that using fallback is not a good idea
even some fallback-ips are blocked.
and some ips are blocked and some not.

patterniha committed:
Update serverless_with_mitm_for_iran.jsonc

I finally decided that using fallback is not a good idea
even some fallback-ips are blocked.
and some ips are blocked and some not.

patterniha committed:
Update README.md

patterniha committed:
Update README.md

patterniha committed:
Update serverless_with_mitm_for_iran.jsonc

dns change to 1.1.1.1

patterniha committed:
Update serverless_for_iran.jsonc

patterniha committed:
Update serverless_with_mitm_for_iran.jsonc

patterniha commented:
in the iran ISP "mokhaberat" fragment is blocked for popular social media youtube, Instagram,...

but MitM works great 

so MitM is useful even now

patterniha committed:
Update serverless_for_iran.jsonc

patterniha committed:
Update serverless_for_iran.jsonc

patterniha committed:
Update serverless_for_iran.jsonc

patterniha committed:
Update serverless_with_mitm_for_iran.jsonc

patterniha committed:
Update serverless_for_iran.jsonc

patterniha committed:
Update serverless_with_mitm_for_iran.jsonc

patterniha committed:
Update README.md

patterniha committed:
Update serverless_for_iran.jsonc

patterniha committed:
Update serverless_with_mitm_for_iran.jsonc

patterniha commented:
Please don't merge for now. There are some issues that need to be handled.

* v2rayNG has a bug in its log.
* Different ISPs show different behavior against fragment
* Different websites show different behavior against fragment

X-Oracle commented:
@patterniha

You mean you are trying to find  a middle ground for different ISPs and websites. Isn't that right?

patterniha commented:
> @patterniha
> 
> You mean you are trying to find a middle ground for different ISPs and websites. Isn't that right?

yes,.
also, the configuration works differently on v2rayNG than directly on Xray-core.
The configs should work on v2rayng as well, because most users use android and v2rayNG.
currently v2rayNG has a bug for serverless configs.
After that bug is fixed, the configs can be merged.

patterniha commented:
it seems @2dust fixed the bug.
i should check everything again then ask @RPRX  to merge

patterniha committed:
Update serverless_for_iran.jsonc

patterniha committed:
Update serverless_with_mitm_for_iran.jsonc

patterniha committed:
Update serverless_for_iran.jsonc

patterniha committed:
Update serverless_with_mitm_for_iran.jsonc

patterniha commented:
@RPRX 
Unfortunately, the serverless features of Xray-core are not working properly on v2rayNG at the moment. I am currently trying to fix it.

patterniha committed:
Update serverless_for_iran.jsonc

add comment

patterniha committed:
Update serverless_with_mitm_for_iran.jsonc

add comment

patterniha committed:
Update serverless_for_iran.jsonc

add comment

patterniha commented:
@Fangliding @RPRX 

i check everything, both works great.
also, the android problem is solved.
Fragment parameters have been optimized for Iran ISPs and MitM is checked and has no problem.

Now please merge into the main.

Thank you for helping the Iranian people for freedom.

patterniha committed:
Update serverless_for_iran.jsonc

fix a minor bug that reported by a user

RPRX commented:
我看到 @patterniha 已经被设为伊朗群管理员了，请给他与 @iambabyninja 完全相同的管理权限

所以 Project X 入群问题的降低难度什么时候安排上 @Fangliding

patterniha commented:
i should add anti-sanction dns ...
It is useful for some websites, although most websites banned based on IP.

Fangliding commented:
给了 都不是啥事那几个权限都不关键
xray无关的问题我都去了 实际上每天都有人答对的((

RPRX commented:
我是想恢复到 Xray-core 的 LICENSE 是 MPL 这种显而易见的问题，Google 就能找到答案的，这已经够门槛了，群需要再次活跃

RPRX commented:
@patterniha 话说你的文件夹，iran 的 i 为什么不大写

patterniha committed:
change iran to Iran

patterniha committed:
iran to Iran

patterniha commented:
> @patterniha 话说你的文件夹，iran 的 i 为什么不大写

done.

With the help of a group of friends, we are creating a list of sites that have banned Iran.

The Iranian government has provided a service called ["shecan"](https://shecan.ir/), which is a DNS that returns its own dedicated IP address for domains that have sanctioned Iran, allowing sanctioned sites to pass through that server and circumvent the sanctions.

This is a free, government-run service.

We just need a list of sanctioned sites, to use "shecan" dns for them.

Please wait a few days, then merge, thank you.

patterniha commented:
@RPRX 
fragment problem is an important bug in Windows that actually hinders the use of these configs in Windows and it must be fixed.

patterniha committed:
Update README.md

RPRX commented:
> Please wait a few days, then merge, thank you.

~~我都打算正式宣布发版了还 wait 吗~~

RPRX commented:
不过我感觉伊朗政府也挺抽象的，一方面致力于封锁一些网站，另一方面又给国民搭（应该是？）端口转发，~~全用代理不行吗~~

话说回来，我觉得这种“绕过美国对伊朗的制裁”的东西不太能放在由美国企业开的网站 GitHub 上，应当在 TG 群传播

“入群获取完整配置”，~~这下还怕伊朗群人少吗~~，你据此修改下 README，说明 Project XHTTP 是伊朗群，“入群获取完整配置”

还有三个文件中的版本要求改为 Xray-core v25.2.21+ @patterniha

patterniha commented:
Ok, I agree.

I don't have access to my pc and Linux right now.

‌I test fragment tonight.

Please give me time to make sure that the fragment has no problem.

RPRX commented:
> Please give me time to make sure that the fragment has no problem.

至少 Xray-core 的代码是没那个问题的，我觉得就是 WireShark 的问题

RPRX committed:
Update README.md

RPRX committed:
Update serverless_for_Iran.jsonc

RPRX committed:
Update serverless_with_mitm_for_Iran.jsonc

RPRX committed:
Update README.md

RPRX committed:
Update serverless_for_Iran.jsonc

RPRX committed:
Update serverless_with_mitm_for_Iran.jsonc

patterniha commented:
@RPRX 
https://github.com/XTLS/Xray-core/issues/4427#issuecomment-2677383324

kanan-droid commented on Serverless-for-Iran/serverless_with_mitm_for_Iran.jsonc:
@RPRX can you send me the .jsonc's content's anywhere is fine
